Podcast/ExternalProfileViewController.swift

```diff
@@ -47,58 +44,68 @@ class ExternalProfileViewController: ViewController, UITableViewDataSource, UITa
         profileTableView.showsVerticalScrollIndicator = false
         mainScrollView = profileTableView
         view.addSubview(profileTableView)
+        profileTableView.snp.makeConstraints { make in
+            make.edges.equalToSuperview()
+        }
         
-        let backgroundExtender = UIView(frame: CGRect(x: 0, y: 0-profileTableView.frame.height+20, width: profileTableView.frame.width, height: profileTableView.frame.height))
+        let backgroundExtender = UIView(frame: CGRect(x: 0, y: 0-view.frame.height+20, width: view.frame.width, height: view.frame.height))
         backgroundExtender.backgroundColor = .sea
         profileTableView.addSubview(backgroundExtender)
         profileTableView.sendSubview(toBack: backgroundExtender)
         
         // Instantiate tableViewHeader and the minified header
-        profileHeaderView = ProfileHeaderView(frame: CGRect(x: 0, y: 0, width: profileTableView.frame.width, height: headerViewHeight))
-        miniHeader = ProfileMiniHeader(frame: CGRect(x: 0, y: 0, width: profileTableView.frame.width, height: miniBarHeight))
-        
+        profileHeaderView = ProfileHeaderView(frame: CGRect(x: 0, y: 0, width: view.frame.width, height: headerViewHeight))
         profileTableView.tableHeaderView = profileHeaderView
         profileHeaderView.delegate = self
         
-        // Add mini header and back button last (make sure it's on top of hierarchy)
+        miniHeader = ProfileMiniHeader(frame: CGRect(x: 0, y: 0, width: view.frame.width, height: miniBarHeight))
         view.addSubview(miniHeader)
+        
+        backButton = UIButton(type: .custom)
+        //backButton.frame = CGRect(x: padding, y: ProfileHeaderView.statusBarHeight + (ProfileHeaderView.miniBarHeight - ProfileHeaderView.statusBarHeight - backButtonHeight) / 2, width: backButtonWidth, height: backButtonHeight)
+        backButton.setImage(UIImage(named: "backArrowLeft"), for: .normal)
+        backButton.contentHorizontalAlignment = .left
+        backButton.adjustsImageWhenHighlighted = true
+        backButton.addTarget(self, action: #selector(didPressBackButton), for: .touchUpInside)
         view.addSubview(backButton)
+        backButton.snp.makeConstraints { make in
+            make.leading.equalToSuperview().inset(padding)
+            make.top.equalToSuperview().inset(ProfileHeaderView.statusBarHeight + (ProfileHeaderView.miniBarHeight - ProfileHeaderView.statusBarHeight - backButtonHeight) / 2)
+            make.width.equalTo(backButtonWidth)
+            make.height.equalTo(backButtonHeight)
+        }
+        
+        loadingAnimation = createLoadingAnimationView()
+        loadingAnimation.center = view.center
+        view.addSubview(loadingAnimation)
+        loadingAnimation.startAnimating()
+        UIApplication.shared.statusBarStyle = .lightContent
         view.bringSubview(toFront: backButton)
+        
+        if let user = user {
+            setUser(user: user)
```

Comment:
might be cleaner if we have the if let inside the setUser/updateViewWithUser functions

Reply:
Maybe. I'll see. I kind of did this order purposefully so you don't accidentally think you're setting it when you have nil.